test/adapter/json_api/linked_test.rb

```diff
@@ -1,11 +1,16 @@
 require 'test_helper'
+
+NestedPost = Class.new(Model)
+class NestedPostSerializer < ActiveModel::Serializer
```

Comment:
I'd say because this model *and* serializer are created here, and specific to this test, they should be namespaced under the test. :-\